src/System.Linq/src/System/Linq/Skip.cs

```diff
@@ -121,5 +122,43 @@ private static IEnumerable<TSource> SkipWhileIterator<TSource>(IEnumerable<TSour
                 }
             }
         }
+
+#if netcoreapp11
+        public static IEnumerable<TSource> SkipLast<TSource>(this IEnumerable<TSource> source, int count)
+        {
+            if (source == null)
+            {
+                throw Error.ArgumentNull(nameof(source));
+            }
+
+            if (count <= 0)
+            {
+                return source.Skip(0);
+            }
+
+            return SkipLastIterator(source, count);
+        }
+
+        private static IEnumerable<TSource> SkipLastIterator<TSource>(IEnumerable<TSource> source, int count)
+        {
+            Debug.Assert(source != null);
+            Debug.Assert(count > 0);
+
+            var queue = new Queue<TSource>();
+
+            using (IEnumerator<TSource> e = source.GetEnumerator())
+            {
+                while (e.MoveNext())
+                {
```

Comment:
It doesn't make much difference there, but it does mean that `queue` will only ever be `count` items in size rather than `count + 1` which if `count` happened to be a growth threshold would be a benefit.

Having the `foreach` broken down allows for a further optimisation of having the loop as:

```C#
while (e.MoveNext())
{
	if (count == 0)
	{
		do
		{
			yield return queue.Dequeue();
			queue.Enqueue(e.Current);
		} while (e.MoveNext());
		break;
	}
	
	count--;
	queue.Enqueue(e.Current);
}
```

Which would avoid testing the count once it had been reached.